parser_library/src/workspaces/workspace.cpp

```diff
@@ -472,66 +487,85 @@ std::regex pathmask_to_regex(std::string input)
     return std::regex(r);
 }
 
-namespace {
-std::string fix_path(std::string path)
-{
-    if (!path.empty() && path.back() != '/' && path.back() != '\\')
-        path.push_back('/');
-    return path;
-};
-} // namespace
-
-void workspace::find_and_add_libs(
-    std::string root, const std::string& path_pattern, processor_group& prc_grp, const library_local_options& opts)
+void workspace::find_and_add_libs(const utils::resource::resource_location& root,
+    const utils::resource::resource_location& path_pattern,
+    processor_group& prc_grp,
+    const library_local_options& opts)
 {
-    std::regex path_validator = pathmask_to_regex(path_pattern);
-
-    std::set<std::string> processed_canonical_paths;
-    std::deque<std::pair<std::string, std::string>> dirs_to_search;
-
-    if (std::error_code ec; dirs_to_search.emplace_back(fix_path(root), utils::path::canonical(root, ec).string()), ec)
+    if (!file_manager_.dir_exists(root))
     {
         if (!opts.optional_library)
-            config_diags_.push_back(diagnostic_s::error_L0001(root));
+            add_diagnostic(diagnostic_s::error_L0001(root.to_presentable()));
         return;
     }
 
+    std::regex path_validator = pathmask_to_regex(path_pattern.get_uri());
+
+    std::set<utils::resource::resource_location> processed_dirs;
+    std::deque<utils::resource::resource_location> dirs_to_search;
```

Comment:
One of the reason why pair was previously used is that one entry tracked the "logical" directory nesting, while the other handled the "physical".
The logical was used to correctly handle relative directory paths, the physical was used to prevent looping. Is this behavior still preserved?

Reply:
Alright, thanks for the insight! 

Nevertheless, I'm not really sure if this can happen anymore as the `root` and `path_pattern` are supposed to be lexically normalized before entering this function. I'll try to think about some tests for this

Reply:
symlinks...